### ISSUE TITLE
Add `prefer-trim` rule

### ISSUE THREAD

**Issue by fir4tozden** [marcalexiei/eslint-plugin-zod#209]
### 🧩 Description

A rule warning against using `z.string().trim()` instead of `z.string()`.

### ⚙️ Options

N/A

### 🧾 Example Configuration

"zod/prefer-trim": "error"

### ✅ Valid Code Examples

```typescript
z.string().trim()
```

### ❌ Invalid Code Examples

```typescript
z.string()
```

**Comment by marcalexiei** [comment]
Rule available is now available in [v3.3.0](https://github.com/marcalexiei/eslint-plugin-zod/releases/tag/v3.3.0).

Thanks for the suggestion!
